crates/core/src/cached_compiler.rs

```diff
@@ -219,21 +196,22 @@ async fn compile_contracts<P: Platform>(
         .collect::<Vec<_>>();
 
     let compiler = Compiler::<P::Compiler>::new()
+        .with_solc_version_req(mode.version.clone())
         .with_allow_path(metadata_directory)
         // Handling the modes
         .with_optimization(mode.optimize_setting)
         .with_pipeline(mode.pipeline)
         // Adding the contract sources to the compiler.
         .try_then(|compiler| {
             files_to_compile.try_fold(compiler, |compiler, path| compiler.with_source(path))
-        })
-        .inspect_err(|err| {
-            compilation_failure_report_callback(
-                Some(compiler_version.clone()),
-                Some(compiler_path.as_ref().to_path_buf()),
-                None,
-                err.to_string(),
-            )
+        // })
```

Comment:
Merging from master introduced a bunch of conflicts mainly related to new reporting stuff. In this file, I've had to comment some of it out to avoid the conflicts, but we need to decide on an approach here now.